records/2024-01-22.reshaping-cicd-environments.md

```diff
@@ -0,0 +1,127 @@
+production release, this would remove some hurdles.
+
+### Safe release to production
+
+The releases need to happen with a certain safety of checks, verifications and
+certified approvals by the change owners. Human interventions might be required
+at certain points to approve of changes or identify issues, potentially
+reverting bad rollouts. At the same time, having too many human intervention
+will lead to a higher number of mistakes.
+
+### Isolated environments for testing
+
+There are changes of certain nature such as infrastructure that leads to
+disruption of the environment itself and might require isolation in a separate
+environment.
+
+## Options
+
+This section lists suggestions made by the team with different considerations
+and goals. We'll first review the current CI/CD process.
+
+### Current process
+
+![Current CI/CD workflow](https://raw.githubusercontent.com/cds-snc/notification-adr/main/records/diagrams/current-cicd.drawio.png?raw=true "Current CI/CD workflow")
+
+### Stronger responsibility for release changes on developers
+
+A lightweight changes that the process can include in the first place is to put
+more burden on the developers' shoulders on the changes approval step.
+Specifically, whatever the developer pushes to the staging environment is what
+will get released by the ops lead.
+
+This keeps the current environments as-is and does not make new automation
+additions. This is a status quo for most release di mensions except for the
+manual aspect.
+
+#### Pros
+
+- Incremental
+- Very cheap to implement
+- Lighten load on the ops lead role
+
+#### Cons
+
+- Higher risk factor of deploying to production (as it relies on multiple human interventions)
+
+### Create dedicated "Core" and "Dev" environments
+
+Currently the staging environment is used as an integration/development environment. This has been workng, but is not aligned with best practices, as it leaves staging in a non-production like state. With the introduction of the new "Dev" environment, it became possible to add an additional environment into the process that allows staging to always remain in the same state as production. 
+
+The new "Dev" envorinment as it exists today is not suitable for such a role because it is constantly being created/destroyed/altered by the Core team in order to test and develop infrastructure changes. It would not be reliable enough for the App team to develop against. 
+
+An additional environment could be created that is reserved for the App team's development and integration, and the existing "Dev" environment could be reclassified as the "Core" environment to identify it as being used by the Core team for infrastructure development.
+
+![Potential Core and Dev Environment Solution](https://raw.githubusercontent.com/cds-snc/notification-adr/main/records/diagrams/2024-01-22.reshaping-cicd-environments/core-dev-env.png?raw=true "Potential Core and Dev Environment Solution")
```

Comment:
same issue as above - the link is not rendering correctly

Reply:
The link is targeting the final version once the PR is done, but you can see the picture attached to this PR though. If we target the link in the PR, it will break once merging into main.